build-system/tasks/runtime-test.js

```diff
@@ -541,96 +547,124 @@ function runTests() {
   const sectionMarker =
       (argv.saucelabs || argv.saucelabs_lite) ? 'saucelabs' : 'local';
 
-  let resolver;
-  const deferred = new Promise(resolverIn => {resolver = resolverIn;});
-  new Karma(c, function(exitCode) {
-    if (shouldCollapseSummary) {
-      console./* OK*/log('travis_fold:end:console_errors_' + sectionMarker);
-    }
-    server.emit('kill');
-    if (exitCode) {
-      log(
-          red('ERROR:'),
-          yellow('Karma test failed with exit code ' + exitCode));
+  // Run Sauce Labs tests in batches to avoid timeouts when connecting to the
+  // Sauce Labs environment.
+  if (argv.saucelabs || argv.saucelabs_lite) {
+    let batch = 1;
+    let startIndex = 0;
+    let endIndex = batchSize;
+    while (startIndex < endIndex) {
+      const configBatch = Object.assign({}, c);
+      configBatch.browsers = saucelabsBrowsers.slice(startIndex, endIndex);
+      log(green('Batch #' + batch + ': Running tests on ' +
+        configBatch.browsers.length + ' Sauce Labs browser(s)...'));
```

Comment:
Add a log statement outside this loop indicating the total number browsers on which tests are being run.

Reply:
Done.